DataCapturing/Source/Persistence/PersistenceLayer.swift

```diff
@@ -334,35 +327,46 @@ public class PersistenceLayer {
-        let rotationsFile = SensorValueFile(fileType: SensorValueFileType.rotationValueType)
-        let directionsFile = SensorValueFile(fileType: SensorValueFileType.directionValueType)
-        if !accelerations.isEmpty {
-            do {
-            _ = try accelerationsFile.write(serializable: accelerations, to: measurement.identifier)
-            } catch {
-                debugPrint("Unable to write data to file \(accelerationsFile.fileName)!")
-                throw error
+            let accelerationsFile = SensorValueFile(fileType: SensorValueFileType.accelerationValueType)
+            let rotationsFile = SensorValueFile(fileType: SensorValueFileType.rotationValueType)
+            let directionsFile = SensorValueFile(fileType: SensorValueFileType.directionValueType)
+            if !accelerations.isEmpty {
+                do {
+                    _ = try accelerationsFile.write(serializable: accelerations, to: measurement.identifier)
+                } catch {
+                    debugPrint("Unable to write data to file \(accelerationsFile.fileName)!")
+                    throw error
+                }
             }
-        }
 
-        if !rotations.isEmpty {
-            do {
-                _ = try rotationsFile.write(serializable: rotations, to: measurement.identifier)
-            } catch {
-                debugPrint("Unable to write data to file \(rotationsFile.fileName)!")
-                throw error
+            if !rotations.isEmpty {
+                do {
+                    _ = try rotationsFile.write(serializable: rotations, to: measurement.identifier)
+                } catch {
+                    debugPrint("Unable to write data to file \(rotationsFile.fileName)!")
+                    throw error
+                }
             }
-        }
 
-        if !directions.isEmpty {
-            do {
-            _ = try directionsFile.write(serializable: directions, to: measurement.identifier)
-            } catch {
-                debugPrint("Unable to write data to file \(directionsFile.fileName)!")
-                throw error
+            if !directions.isEmpty {
+                do {
+                    _ = try directionsFile.write(serializable: directions, to: measurement.identifier)
+                } catch {
+                    debugPrint("Unable to write data to file \(directionsFile.fileName)!")
+                    throw error
+                }
             }
+
+            measurement.accelerationsCount = measurement.accelerationsCount.advanced(by: accelerations.count)
+            measurementMO.accelerationsCount = measurement.accelerationsCount
+            measurement.rotationsCount = measurement.rotationsCount.advanced(by: rotations.count)
+            measurementMO.rotationsCount = measurement.rotationsCount
+            measurement.directionsCount = measurement.directionsCount.advanced(by: accelerations.count)
+            measurementMO.directionsCount = measurement.directionsCount
```

Comment:
true. But probably not important. We are not using those counts anymore. It just stays here, since If I remove this I would have to change the database schema and provide new migration code. As I am not sure how stable that migration code is, I would like to wait with that until September.